### ISSUE TITLE
Mongodump issue

### ISSUE THREAD

**Issue by gaboluque** [llangit/lambda-mongocluster-s3#16]
I keep getting this error:  
```
Error: mongodump command failed: 
    at handler (file:///.../lambda-mongocluster-s3/index.js:31:11)
```

Any ideas?

**Comment by ashishmeher216** [comment]
I am also getting the same error. Did you get any solution?

**Comment by llangit** [comment]
Could it be that the network access settings of your MongoDB instance block access from your Lambda function? I get the same error when that is the case.

**Comment by ashishmeher216** [comment]
I don’t think so. The atlas is accessible. The problem is with mongodump. When I install mongodump in ec2 and fire the request it works. In the lambda case I tried changing the machine architecture as well but no luck.

**Comment by llangit** [comment]
1. So your Lambda function is on x86_64 and Node 18.x?
2. You have increased the memory to at least 512, ephemeral storage to at least 1024 and the timeout to at least 1min?
3. How large are your dump files if you mongodump outside of Lambda?

**Comment by ashishmeher216** [comment]
1. Yes
2. Yes
3. Totalling around 3 MB

![image](https://github.com/llangit/lambda-mongocluster-s3/assets/51280607/1e674ef9-7bef-40d8-96c8-4652231d7845)
![image](https://github.com/llangit/lambda-mongocluster-s3/assets/51280607/e5cf1817-4be9-48b6-a109-72de8555eb80)

Also, the max memory utilization during lambda execution is around 115MB.

**Comment by llangit** [comment]
Ok good. And what do you have in MONGODUMP_OPTIONS? Only the --uri in the format "mongodb+srv://[user]:[pass]@[host]/[name]" or other arguments as well?
And just to make sure: you're using the current version of the repo, right?

**Comment by ashishmeher216** [comment]
Yes, I am using the latest version of the repo. I only have mongodb uri in the MONGODUMP_OPTIONS as mentioned in the readme file.

**Comment by llangit** [comment]
Ok, then everything should be set up correctly.
While trying to reproduce your issue this morning, I've created a new Lambda function in eu-central-1, configured it as discussed above, uploaded the files through the AWS interface, and tested the function through the "Test" tab (hello-world template). It worked seamlessly for me.
Based on the information you have given me, I'm not really sure where your issue might be coming from.

**Comment by ashishmeher216** [comment]
Below is the snapshot of the index.js file. Can you please check if there is any problem here? Env variables are correctly configured in the lambda.

![image](https://github.com/llangit/lambda-mongocluster-s3/assets/51280607/bfa43fe2-8c48-42ad-9408-b52dec1f8270)

**Comment by llangit** [comment]
Code looks good to me - can't find any error.
What version is your MongoDB cluster on? Are you using Serverless, Dedicated or Shared? Is it a 3-node replica set?
You could also try to replace the binary by the newest version 100.9.0 (https://www.mongodb.com/try/download/database-tools)

**Comment by ashishmeher216** [comment]
MongoDB is on M0 cluster, shared, 3-node replica set. I will try to replace the mongodump binary.
